### PR TITLE
Add Jenkins Job for Validation workflow

### PR DESCRIPTION
**Pull request by Divyaasm**
### Description

Created jenkins Job for Validation Workflow

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: jenkins/validate-artifacts/validate-artifacts.jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+pipeline {
+    options {
+        timeout(time: 1, unit: 'HOURS')
+    }
+    agent none
+    parameters {
+        string(
+            name: 'VERSION',
+            description: 'Enter the OpenSearch and OpenSearch Dashboards VERSION that needs to be validated',
```
Comment (gaiksaya) [inline jenkins/validate-artifacts/validate-artifacts.jenkinsfile:36]:
```suggestion
            description: 'Enter the OpenSearch and OpenSearch Dashboards version that needs to be validated',
```

File: jenkins/validate-artifacts/validate-artifacts.jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+            trim: true
+        )
+        string(
+            name: 'DISTRIBUTION',
+            description: 'Enter the distribution type among [tar,rpm,yum,docker] default is set to All distributions',
```
Comment (gaiksaya) [inline jenkins/validate-artifacts/validate-artifacts.jenkinsfile:41]:
```suggestion
            description: 'Enter the distribution type among tar, rpm, yum, docker separated by space. Defaults to "tar rpm yum docker"',
```

File: jenkins/validate-artifacts/validate-artifacts.jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+            trim: true,
+            defaultValue: "tar rpm yum docker"
+
+        )
+        string(
+            name: 'ARCHITECTURE',
+            description: 'Enter the architecture type among x64 and arm64 default is set to [x64,arm64]',
```
Comment (gaiksaya) [inline jenkins/validate-artifacts/validate-artifacts.jenkinsfile:48]:
```suggestion
            description: 'Enter the architecture type among x64 and arm64 separated by space. Defaults "x64,arm64"',
```

File: jenkins/validate-artifacts/validate-artifacts.jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+            trim: true,
+            defaultValue: "x64 arm64"
+        )
+        string(
+            name: 'OS_BUILD_NUMBER',
+            description: 'Enter the build number for opensearch - This parameter is applicable only for Docker Validation',
```
Comment (gaiksaya) [inline jenkins/validate-artifacts/validate-artifacts.jenkinsfile:54]:
```suggestion
            description: 'Enter the build number for OpenSearch - This parameter is applicable only for Docker Validation',
```

File: jenkins/validate-artifacts/validate-artifacts.jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+            trim: true,
+        )
+        string(
+            name: 'OSD_BUILD_NUMBER',
+            description: 'Enter the build number for opensearch Dashboards - This parameter is applicable only for Docker Validation',
```
Comment (gaiksaya) [inline jenkins/validate-artifacts/validate-artifacts.jenkinsfile:59]:
```suggestion
            description: 'Enter the build number for OpenSearch Dashboards - This parameter is applicable only for Docker Validation',
```

File: jenkins/validate-artifacts/validate-artifacts.jenkinsfile [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+            trim: true,
+        )
+        string(
+            name: 'OPTIONAL_ARGS',
+            description: 'Choose among validate-digest-only or using-staging-artifact-only [Optional]',
```
Comment (gaiksaya) [inline jenkins/validate-artifacts/validate-artifacts.jenkinsfile:64]:
```suggestion
            description: 'Optional args for docker validation. Currently supports validate-digest-only or using-staging-artifact-only. Defaults to using-staging-artifact-only',
```

File: jenkins/validate-artifacts/validate-artifacts.jenkinsfile [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+            agent { label agent_nodes["x64"] }
+            steps {
+                script {
+                    if (VERSION == '') {
+                        currentBuild.result = 'ABORTED'
+                        error('Provide version number for the artifacts')
+                    }
+                    if (params.DISTRIBUTION.contains('docker') && (params.OS_BUILD_NUMBER == '' || params.OSD_BUILD_NUMBER == '')) {
+                        currentBuild.result = 'ABORTED'
+                        error("Provide OS_BUILD_NUMBER and OSD_BUILD_NUMBER for Docker Validation")
```
Comment (gaiksaya) [inline jenkins/validate-artifacts/validate-artifacts.jenkinsfile:82]:
```suggestion
                        error("Provide OS_BUILD_NUMBER and OSD_BUILD_NUMBER args for Docker Validation")
```

File: jenkins/validate-artifacts/validate-artifacts.jenkinsfile [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+    }
+
+    stages {
+        stage('verify-parameters') {
+            agent { label agent_nodes["x64"] }
+            steps {
+                script {
+                    if (VERSION == '') {
+                        currentBuild.result = 'ABORTED'
+                        error('Provide version number for the artifacts')
```
Comment (gaiksaya) [inline jenkins/validate-artifacts/validate-artifacts.jenkinsfile:78]:
```suggestion
                        error('VERSION cannot be empty. Please provide version of the artifacts to validate')
```

File: tests/jenkins/TestValidateArtifacts.groovy [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+    public void testValidateArtifactsPipeline() {
+        super.testPipeline("jenkins/validate-artifacts/validate-artifacts.jenkinsfile",
+                "tests/jenkins/jenkinsjob-regression-files/validate-artifacts/validate-artifacts.jenkinsfile")
+        assertThat(getCommandExecutions('sh', 'validation.sh'), hasItem('/tmp/workspace/validation.sh  --version 2.3.0 --distribution docker --arch x64 --os-build-number 6039 --osd-build-number 4104 --using-staging-artifact-only '))
+        assertThat(getCommandExecutions('sh', 'validation.sh'), hasItem('/tmp/workspace/validation.sh  --version 2.3.0 --distribution tar --arch arm64'))
+        assertThat(getCommandExecutions('sh', 'validation.sh'), hasItem('/tmp/workspace/validation.sh  --version 2.3.0 --distribution rpm --arch x64'))
+        assertThat(getCommandExecutions('sh', 'validation.sh'), hasItem('/tmp/workspace/validation.sh  --version 2.3.0 --distribution yum --arch arm64'))
+
+    }
+
```
Comment (gaiksaya) [inline tests/jenkins/TestValidateArtifacts.groovy:62]: Add one more test to throw an error in case docker args are not provided.

**Review by gaiksaya — CHANGES_REQUESTED** [review]

**Review by gaiksaya — APPROVED** [review]

**Comment by prudhvigodithi** [comment]
Hey @Divyaasm can you please take of the NIT's, we can proceed to merge this PR.
Thank you

**Comment by gaiksaya** [comment]
I think we are good @prudhvigodithi The nits are showing as `outdated`

**Pull request merged**